edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/planners/HealthRestartPlanner.cs

```diff
@@ -393,6 +399,29 @@ async Task<List<ICommand>> GetUpdateRuntimeCommands(IList<IModule> updateDeploye
             return updateRuntimeCommands;
         }
 
+        async Task<IEnumerable<ICommand>> GetUpdateSecretsCommands(IEnumerable<IModule> desiredModules, IEnumerable<IModule> currentModules)
+        {
+            IList<(IModule, string, string)> desiredSecrets = desiredModules
```

Comment:
I agree that we should not compare all properties. However, I think we should verify that, for example, the image has not changed since the previous deployment.